modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -928,11 +928,32 @@ def _reorder_labels(self, row_positions=None, col_positions=None):
         PandasDataframe
             A new PandasDataframe with reordered columns and/or rows.
         """
+        new_row_lengths = None
+        new_column_widths = None
+        if row_positions is not None and col_positions is None:
+            # op doesn't affect another axis
+            new_column_widths = self._column_widths_cache
+        if row_positions is None and col_positions is not None:
+            # op doesn't affect another axis
+            new_row_lengths = self._row_lengths_cache
         if row_positions is not None:
             ordered_rows = self._partition_mgr_cls.map_axis_partitions(
                 0, self._partitions, lambda df: df.iloc[row_positions]
             )
             row_idx = self.index[row_positions]
+            if self._row_lengths_cache is not None and sum(self._row_lengths_cache) == len(row_idx):
+                from modin.config import NPartitions
+                if NPartitions.get() == len(ordered_rows):
```

Comment:
```suggestion
                n_parts = NPartitions.get()
                if NPartitions.get() == len(ordered_rows):
```
let's move the import at the header and make a var